arch/arm/src/Makefile

```diff
@@ -80,6 +80,10 @@ endif
 # The "head" object
 
 HEAD_OBJ = $(HEAD_ASRC:.S=$(OBJEXT))
+ifneq ($(STARTUP_OBJS),)
+  # Assume C by default
+  STARTUP_COBJS ?= $(STARTUP_OBJS)
+endif
 STARTUP_OBJS ?= $(HEAD_OBJ)
```

Comment:
How about we remove line 83 to 87, and put $(STARTUP_OBJS) to line 175 directly? it doesn't make sense to assign HEAD_OBJ to STARTUP_OBJS if it is empty because HEAD_OBJ is for kernel startup and never suitable for userspace startup.
Also, we need modify other arch to ensure crt0.c get compiled.

Reply:
isn't it necessary for non KERNEL configs?

Reply:
All HEAD_ASRC has the code to initialize the whole system, it's always not suitable as raw application enrty point in any build mode.

Reply:
my understanding/assumption was that:
* the export target is to collect things necessary to build and link apps
* in case of non-KERNEL builds, it can includes HEAD_OBJ

Reply:
> my understanding/assumption was that:
> 
> * the export target is to collect things necessary to build and link apps

Yes, export target is just for apps initialization.

> * in case of non-KERNEL builds, it can includes HEAD_OBJ

No, because HEAD_OBJ contain the machine start up code, launching apps linked with HEAD_OBJ most likely will generate the panic in any build mode. Can you imagine the result that HEAD_OBJ code(initialize the machine) run more than once?

Reply:
well, for non-KERNEL builds, "linking apps" involves the whole kernel, doesn't it?

Reply:
anyway, it's entirely possible i don't understand "make export" is. i haven't successfully used it. i tried it just because it was in the procedure in sema5 README.

Reply:
> well, for non-KERNEL builds, "linking apps" involves the whole kernel, doesn't it?

Yes, if apps is built into kernel. But if we build apps as elf binary, it is separated from kernel totally.

> anyway, it's entirely possible i don't understand "make export" is. i haven't successfully used it. i tried it just because it was in the procedure in sema5 README.

I also don't use export before, from my understanding: export is used for 3rd party develop application without full nuttx source code(something like Android NDK).